adapters/httpapi_compact.c

```diff
@@ -116,16 +122,16 @@ static int ParseStringToHexadecimal(const char *src, size_t* dst)
     return result;
 }
 
-/*the following function does the same as sscanf(buf, "HTTP/%*d.%*d %d %*[^\r\n]", &ret) */
+/*the following function does the same as sscanf(buf, "HTTP/%*d.%*d %d %s[^\r\n]", &code, reasonPhrase) */
 /*this function only exists because some of platforms do not have sscanf. This is not a full implementation; it only works with well-defined HTTP response. */
-static int  ParseHttpResponse(const char* src, int* dst)
+static int  ParseHttpResponse(const char* src, int* code, char* reasonPhrase)
```

Comment:
need add one more parameter telling length of the buffer, otherwise there could be buffer over run